### PR TITLE
Chore: bump from lts-18.28 to 19.0

### PR DESCRIPTION
**Pull request by mitchellwrosen**
- [x] Depends on #3629 
- [x] Depends on #3630 
- [x] Depends on #3633 

---

## Overview

This PR bumps the Stackage snapshot from `lts-18.28` to `lts-19.0`. I chose `lts-19.0` as it's the first snapshot after `lts-18.28` that bumps the GHC version (from `8.10.7` to `9.0.2`). I figured it would be easier to land here than all the way at the latest snapshot (currently `lts-20.1`, on GHC `9.2.5`).

### CODE REVIEW

File: codebase2/core/U/Core/ABT.hs [hunk 1]
```diff
@@ -44,7 +46,7 @@ instance
   ( forall a. Eq a => Eq (f a),
     Foldable f,
     Functor f,
-    forall a. Ord a => Ord (f a),
```
Comment (mitchellwrosen) [inline codebase2/core/U/Core/ABT.hs:47]: This one was weird. GHC 9.0 fixed a bug in `QuantifiedConstraints` that this line triggered; I played around a bit but couldn't figure out how to get the thing compiling with the quantified constraint on `Ord`, so I replaced it with an older-style `Ord (f (Term f v a))` + `UndecidableInstances`. It compiles, so I think it works. Here's a relevant StackOverflow question https://stackoverflow.com/questions/68956257/change-in-behaviour-of-quantified-constraints-in-ghc-9

**Comment by aryairani** [comment]
Is there an hls for this 9.0.2?

**Comment by ChrisPenner** [comment]
I'll try pulling this and test out HLS and such, reminder to test Enlil against this snapshot too pls

**Comment by ChrisPenner** [comment]
Note: This build does trigger this GHC 9.0.2 bug on M1 macs: (even though it's reported against a different version) https://gitlab.haskell.org/ghc/ghc/-/issues/20592

This workaround does the trick, kind of annoying to prefix every command though haha;

https://gitlab.haskell.org/ghc/ghc/-/issues/20592#note_391266

**Comment by mitchellwrosen** [comment]
Hrm; good catch. Ok, then we should skip this and hit 9.2.3 or later, where that bug no longer exists.

**Comment by aryairani** [comment]
So, close?

**Comment by mitchellwrosen** [comment]
Ya

**Pull request closed without merging**